homeassistant/components/geonetnz_volcano/sensor.py

```diff
@@ -0,0 +1,169 @@
+        if self._remove_signal_update:
+            self._remove_signal_update()
+
+    @callback
+    def _update_callback(self):
+        """Call update method."""
+        self.async_schedule_update_ha_state(True)
+
+    @property
+    def should_poll(self):
+        """No polling needed for GeoNet NZ Volcano feed location events."""
+        return False
+
+    async def async_update(self):
+        """Update this entity from the data held in the feed manager."""
+        _LOGGER.debug("Updating %s", self._external_id)
+        feed_entry = self._feed_manager.get_entry(self._external_id)
+        last_update = self._feed_manager.last_update()
+        last_update_successful = self._feed_manager.last_update_successful()
+        if feed_entry:
+            self._update_from_feed(feed_entry, last_update, last_update_successful)
+
+    def _update_from_feed(self, feed_entry, last_update, last_update_successful):
+        """Update the internal state from the provided feed entry."""
+        self._title = feed_entry.title
+        # Convert distance if not metric system.
+        if self._unit_system == CONF_UNIT_SYSTEM_IMPERIAL:
+            self._distance = round(
+                IMPERIAL_SYSTEM.length(feed_entry.distance_to_home, LENGTH_KILOMETERS),
+                1,
+            )
+        else:
+            self._distance = round(feed_entry.distance_to_home, 1)
+        self._latitude = round(feed_entry.coordinates[0], 5)
+        self._longitude = round(feed_entry.coordinates[1], 5)
+        self._attribution = feed_entry.attribution
+        self._alert_level = feed_entry.alert_level
+        self._activity = feed_entry.activity
+        self._hazards = feed_entry.hazards
+        self._feed_last_update = dt.as_utc(last_update) if last_update else None
+        self._feed_last_update_successful = (
+            dt.as_utc(last_update_successful) if last_update_successful else None
+        )
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self._alert_level
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return DEFAULT_ICON
+
+    @property
+    def name(self) -> Optional[str]:
+        """Return the name of the entity."""
+        return f"Volcano {self._title}"
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return "alert level"
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {}
+        for key, value in (
+            (ATTR_EXTERNAL_ID, self._external_id),
+            (ATTR_ATTRIBUTION, self._attribution),
+            (ATTR_ACTIVITY, self._activity),
+            (ATTR_HAZARDS, self._hazards),
+            (ATTR_LONGITUDE, self._longitude),
+            (ATTR_LATITUDE, self._latitude),
+            (ATTR_DISTANCE, self._distance),
```

Comment:
We should create a geo_location entity for the distance value. Keep the sensor platform for the alert level. We prefer many different entities over one entity with many different attributes.

Reply:
I see we don't remove entities. Then the geo_location entity will never change unless the user changes radius. So maybe it's not needed? It still good to get the distance as a sensor state though. 🤷‍♂️

Reply:
Let's keep it as an attribute. You were right the first time. :+1: